src/Workspaces/Core/Portable/Execution/Asset.cs

```diff
@@ -121,7 +121,7 @@ internal sealed class AnalyzerReferenceAsset : Asset
         private readonly AnalyzerReference _reference;
 
         public AnalyzerReferenceAsset(Serializer serializer, AnalyzerReference reference, Checksum checksum, string kind) :
-            base(checksum, WellKnownChecksumObjects.AnalyzerReference)
+            base(Checksum.Create(kind, checksum), WellKnownChecksumObjects.AnalyzerReference)
         {
             Contract.Requires(kind == WellKnownChecksumObjects.AnalyzerReference);
```

Comment:
I don't really understand this answer.  Why do you require a value to be passed in... when you already know what that value must be.   Why not remove the 'kind' parameter entirely and just use "WellKnownChecksumObjects.AnalyzerReference" entirely in the body of this constructor?